app/controllers/api/conversion_hosts_controller.rb

```diff
@@ -1,5 +1,79 @@
 module Api
   class ConversionHostsController < BaseController
     include Subcollections::Tags
+
+    # Whitelist of valid resource types
+    VALID_TYPES = {
+      'ManageIQ::Providers::Openstack::CloudManager::Vm' => ManageIQ::Providers::Openstack::CloudManager::Vm,
+      'ManageIQ::Providers::Redhat::InfraManager::Host'  => ManageIQ::Providers::Redhat::InfraManager::Host
```

Comment:
I really don't like putting the provider constants in this code for 2 reasons.. 1) pluggability...perhaps this can be done with a registrations of some sort and 2) This is defined at the class level, so will force resolutions of this const at boot time.

Reply:
I could use the `.supports?` method, but that of course means I have to constantize first.

Reply:
as long as the constantize happens at execution time and not boot time, I'm fine with that.

Reply:
Alternately, you can traverse all provider plugins with `Vmdb::Plugins.provider_plugins.each`, so no need to call constantize.

Reply:
I think scanning the entire class hierarchy for every provider is simply not practical. Instead, I've used a combination of `safe_constantize` and the baked in `supports_conversion_host?` logic to validate the params. I think this is as good, safe, and flexible as we can be.